api/v1alpha1/validator/nicclusterpolicy_webhook.go

```diff
@@ -474,6 +484,168 @@ func validateRepository(repo string, allErrs field.ErrorList, fp *field.Path, ch
+			states["cniPlugins"] = stateRenderData{
+				policy.Spec.SecondaryNetwork.CniPlugins, state.NewStateCNIPlugins,
+				filepath.Join(manifestBaseDir, "state-container-networking-plugins"),
+			}
+		}
+		if policy.Spec.SecondaryNetwork.IPoIB != nil {
+			states["ipoib"] = stateRenderData{
+				policy.Spec.SecondaryNetwork.IPoIB, state.NewStateIPoIBCNI,
+				filepath.Join(manifestBaseDir, "state-ipoib-cni"),
+			}
+		}
+		if policy.Spec.SecondaryNetwork.Multus != nil {
+			states["multus"] = stateRenderData{
+				policy.Spec.SecondaryNetwork.Multus, state.NewStateMultusCNI,
+				filepath.Join(manifestBaseDir, "state-multus-cni"),
+			}
+		}
+		if policy.Spec.SecondaryNetwork.IpamPlugin != nil {
+			states["ipamPlugin"] = stateRenderData{
+				policy.Spec.SecondaryNetwork.IpamPlugin, state.NewStateWhereaboutsCNI,
+				filepath.Join(manifestBaseDir, "state-whereabouts-cni"),
+			}
+		}
+		for stateName, renderData := range states {
+			localData := renderData
+			allErrs = validateContainerResourcesIfNotNil(&localData, policy, allErrs, snfp, stateName)
+		}
+	}
+	return allErrs
+}
+
+func validateContainerResourcesIfNotNil(
+	resource *stateRenderData, policy *v1alpha1.NicClusterPolicy, allErrs field.ErrorList,
+	fp *field.Path, fieldName string) field.ErrorList {
+	if resource.provider == nil {
+		return allErrs
+	}
+
+	_, renderer, err := resource.newState(nil, nil, resource.manifestDir)
+	if err != nil {
+		nicClusterPolicyLog.Error(err, "failed to created state renderer")
+		return allErrs
+	}
+	return validateResourceRequirements(
+		resource.provider.GetContainerResources(), policy, renderer, allErrs, fp, fieldName)
+}
+
+func validateResourceRequirements(resources []v1alpha1.ResourceRequirements,
+	policy *v1alpha1.NicClusterPolicy, manifestRenderer state.ManifestRenderer,
+	allErrs field.ErrorList, fp *field.Path, child string) field.ErrorList {
+	if len(resources) == 0 {
+		return allErrs
+	}
+
+	supportedContainerNames, err := state.ParseContainerNames(manifestRenderer, policy, nicClusterPolicyLog)
+	if err != nil {
+		nicClusterPolicyLog.Error(err, "failed to parse container names")
+	}
+
+	for _, reqs := range resources {
+		allErrs = validateResources(reqs.Requests, allErrs, fp, child, "Requests")
+		allErrs = validateResources(reqs.Limits, allErrs, fp, child, "Limits", reqs.Requests)
+		if !slices.Contains(supportedContainerNames, reqs.Name) {
+			allErrs = append(
+				allErrs, field.Invalid(fp.Child(child).Child("containerResources").Child("name"),
+					reqs.Name, fmt.Sprintf("container name %s is unsupported for this state", reqs.Name)))
+		}
+	}
+
+	return allErrs
+}
+
+func validateResources(resources map[v1.ResourceName]apiresource.Quantity, allErrs field.ErrorList, fp *field.Path,
+	child, resourceType string, requests ...map[v1.ResourceName]apiresource.Quantity) field.ErrorList {
+	for resourceName, quantity := range resources {
+		if resourceName == v1.ResourceCPU || resourceName == v1.ResourceMemory {
```

Comment:
i believe we want to limit req/limits to just CPU and Memory no ?

if its something else which was provided we fail, WDYT ?

Reply:
Let’s do it. We probably don’t care about other resources

Reply:
Done